src/compiler/transformers/utilities.ts

```diff
@@ -34,7 +34,8 @@ namespace ts {
     }
 
     export function getImportNeedsImportStarHelper(node: ImportDeclaration) {
-        return !!getNamespaceDeclarationNode(node) || (getNamedImportCount(node) > 1 && containsDefaultReference(node.importClause.namedBindings));
+        const minNamed = isDefaultImport(node) ? 0 : 1;
+        return !!getNamespaceDeclarationNode(node) || (getNamedImportCount(node) > minNamed && (!minNamed || containsDefaultReference(node.importClause.namedBindings)));

```

Comment:
this assumes that `default` will only appear at most once in the named list, but nothing stops you from writing `import {default as a, default as b} from "m";`. which should be harmless. but worth adding a test.